unit_tests/test_decode_backtrace.py

```diff
@@ -190,3 +195,142 @@ def test_04_decode_backtraces_core(self):
             if event.get('backtrace') and event.get('raw_backtrace'):
                 self.assertEqual(event['backtrace'].strip(),
                                  "addr2line -Cpife scylla_debug_info_file {}".format(' '.join(event['raw_backtrace'].split("\n"))))
+
+
+# Pytest-style fixtures and tests
+
+@pytest.fixture
+def test_config_fixture(events_function_scope):
+    """Fixture to create TestConfig with decoding queue."""
+    config = TestConfig()
+    config.set_decoding_queue()
+    config.BACKTRACE_DECODING = True
+    config.DECODING_QUEUE = Queue()
+    yield config
+
+
+@pytest.fixture
+def dummy_node_fixture(tmp_path):
+    """Fixture to create a dummy node for testing."""
+    dummy_node = DecodeDummyNode(
+        name='test_node',
+        parent_cluster=None,
+        base_logdir=tmp_path,
+        ssh_login_info=dict(key_file='~/.ssh/scylla-test'),
+    )
+    dummy_node.remoter = DummyRemote()
+    dummy_node.system_log = os.path.join(os.path.dirname(__file__), 'test_data', 'system.log')
+    return dummy_node
+
+
+@pytest.fixture
+def monitor_node_fixture(tmp_path):
+    """Fixture to create a monitor node for testing."""
+    monitor_node = DecodeDummyNode(
+        name='test_monitor_node',
+        parent_cluster=None,
+        base_logdir=tmp_path,
+        ssh_login_info=dict(key_file='~/.ssh/scylla-test'),
+    )
+    monitor_node.remoter = DummyRemote()
+    yield monitor_node
+    # Cleanup
+    try:
+        monitor_node.termination_event.set()
+        monitor_node.stop_task_threads()
+        monitor_node.wait_till_tasks_threads_are_stopped()
+    except Exception:  # noqa: BLE001
+        pass  # Best effort cleanup
```

Comment:
We are in test scenario, how can this fail, we should have control over everything